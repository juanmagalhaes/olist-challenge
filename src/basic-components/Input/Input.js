// @flow

import * as React from 'react';
import styles from './Input.module.scss';
import classnames from 'classnames';

type Props = {
  type?: string,
  className?: string,
}

const Input = ({
  type="text",
  className='',
  ...rest,
}: Props) => (
  <input
    {...rest}
    type={type}
    className={classnames(
      styles.input,
      className
    )}
  />
);

export { Input };

