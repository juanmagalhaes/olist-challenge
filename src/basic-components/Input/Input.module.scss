@import '~global-styles/variables';

$input-width: 357px;
$input-height: 44px;
$input-padding: 0 8px;
$input-color: #312F4F;
$input-font-size: 14px;
$input-border-color: #B6B9D0;
$input-border: solid 1px $input-border-color;
$input-inner-shadow: inset 0 0 3px rgba(0, 0, 0, 0.05);

.input {
  color: $input-color;
  width: $input-width;
  height: $input-height;
  border: $input-border;
  padding: $input-padding;
  font-size: $input-font-size;
  box-shadow: $input-inner-shadow;

  &Default {
    border-color: $input-border-color;
  }

  &Danger {
    border-color: $danger-bg;
    &:focus { outline: none; }
  }

  &Waring {
    border-color: $waring-bg;
    &:focus { outline: none; }
  }

  &Success {
    border-color: $success-bg;
    &:focus { outline: none; }
  }
}
