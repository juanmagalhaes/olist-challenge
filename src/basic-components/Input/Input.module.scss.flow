// @flow
/* This file is automatically generated by css-modules-flow-types */
declare module.exports: {|
  +'input': string;
  +'inputDanger': string;
  +'inputDefault': string;
  +'inputSuccess': string;
  +'inputWaring': string;
|};
