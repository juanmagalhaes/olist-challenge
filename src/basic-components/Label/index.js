// @flow

export { Label } from './Label';

