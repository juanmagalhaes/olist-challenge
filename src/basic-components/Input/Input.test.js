import React from 'react';
import { Input } from '~/basic-components/Input';

test('Input text', () => expect(shallow(<Input />)).toMatchSnapshot())

