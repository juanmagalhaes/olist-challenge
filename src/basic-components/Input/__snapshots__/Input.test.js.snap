// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input text 1`] = `
<input
  className="input"
  type="text"
/>
`;
